### PR TITLE
Capture the Flag

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
-Flags. You can't pick up your own flag. Flags require 2 hands to carry and slow you down, so your team has to cover you. They give Z level wide messages when picked up/dropped. They return to their spawn point after 20 seconds of not being held.

-Team control machines. Clicking as a ghost spawns you in with a a shielded hardsuit/autorifle/deagle. The machine checks to make sure the other teams have enough players, that you're not on a different team, and that your last death wasn't within 15 seconds. Hitting it with another teams flag (you could have theoretically infinite teams) scores. 3 points and you win/the machines reset. Admins can enable/disable the machines.

-A shitty map to play CTF on for christmas. Loaded as an away mission, but has no gate.

-Spawns are protected by divine trap runes that dust you if you're on the wrong team

I have not been able to test this yet because its kind of a pain with just one person (I'll do it in a bit but still)

### CODE REVIEW

**Comment by Supermichael777** [comment]
Red team op pls nerf.

IF it works cool.

**Comment by Bawhoppen** [comment]
I don't like the idea of it being scripted and forced to work a certain way, This game is about players and their interactions as they were the characters, not game mechanics making it work a certain way (within plain sight)
like arbitrarily not being able to pick up your own flag
You should be able to do things even if they break the rules of that game (in which case you have your rule enforcer with a sniper rifle to deal with the trouble makers)

File: code/modules/awaymissions/mission_code/capture_the_flag.dm [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+		return
+
+	pickup(user)
+	for(var/mob/M in player_list)
+		if (M.z == src.z)
+			M << "<span class='userdanger'>/The [src] has been taken!</span>"
+	if(!user.put_in_active_hand(src))
+		dropped(user)
+		return
+	for(var/mob/M in player_list)
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: uh... you report this twice?

Comment (KorPhaeron) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: Its really important or something

File: code/modules/awaymissions/mission_code/capture_the_flag.dm [hunk 2]
```diff
@@ -0,0 +1,238 @@
+#define WHITE_TEAM "White"
+#define RED_TEAM "Blue"
+#define BLUE_TEAM "Red"
+#define CTF_RESPAWN_COOLDOWN 150 // 15 seconds
+#define FLAG_RETURN_TIME 200 // 20 seconds
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: you never use this.

**Comment by KorPhaeron** [comment]
This is for a specific holiday event/thunderdome style admin run death matches, not the normal game

File: code/modules/awaymissions/mission_code/capture_the_flag.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: But I should, in the process. Will fix in a bit

Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: I know you should, but as I said, you never do.

**Comment by Robustin** [comment]
Seems like it would work better with non-lethals rather than forcing me to suicide every time I fall below 60hp and start moving at cripple speed, specifically taser-style weaponry (because, again, a full stamina stun would leave you moving slower for like 90 seconds, I would consider just ghosting once I collapsed from disabler spam).

Tasers serve their purpose without leaving the field littered with corpses/guns.

Ideally it would be auto-recharging tasers that use 2-hands to fire (so that looting another gun doesn't turn you into a double taser turbohitler), and have a more significant delay between shots (sorry kids time to aim and not just spray electrodes down the hall). Bonus points if you make it so that the taser effect doesn't work on downed enemy (so you can't just permastun an enemy).

Also there should be a mechanic to prevent teamstacking, ideally there's just one "join" machine and it plops you on the least populated team. Also touching your own flag should immediately teleport it back.

A few simple rules upon spawning would be nice, i.e.

1) No stealing weapons
2) No harm

What you have right now is more akin to those shitty highlander rounds where respawning is enabled. With 15s respawn and high powered ballistic weaponry both sides will be capable of becoming virtually impenetrable and it's just a silly TDM without any realistic win condition.

**Comment by YotaXP** [comment]
I'd say make players gib the moment they fall unconscious.

File: code/modules/awaymissions/mission_code/capture_the_flag.dm [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+					M << "<span class='userdanger'>[user.real_name] has captured /the [src], scoring a point for [team] team! They now have [points]/[points_to_win] points!</span>"
+		if(points == points_to_win)
+			victory()
+
+
+
+/obj/machinery/capture_the_flag/proc/victory()
+	for(var/mob/M in player_list)
+		if (M.z == src.z)
+			M << "<span class='narsie'>[team] team wins!</span>"
```
Comment (Anonus) [inline code/modules/awaymissions/mission_code/capture_the_flag.dm]: MAKE A NEW SPAN CLASS YOU NERD DON'T USE THIS ONE
HECK, PASS THE WINNING TEAM'S COLOR DOWN TO THIS SO YOU CAN MAKE IT ANY COLOR

**Comment by Shadowlight213** [comment]
@KorPhaeron We probably could use this system as a framework for the ghost Valhalla idea that was proposed a long time ago. Like little deathmatch arenas for bored ghosts to fight in during long rounds.

**Comment by Aranclanos** [comment]
it feels like you're using ss13 as an engine to make a game on top of it, its just so weird.
the ckey team handling is horrible and the user messages and gameplay are completly ooc, have nothing to do with ss13 and break all roleplay.
i mean, you really made a different game using ss13 as an engine.

**Comment by Steelpoint** [comment]
Make it like the training section in Starship Troopers, mainly in that if you take a single hit your "killed". This would encourage people to be more careful in their approach.

**Comment by Firecage** [comment]
Oh, and here I thought you were actually re-adding the actually fun old CTF mode which SS13 had at a time. No thanks.

**Comment by KorPhaeron** [comment]
Not meant to be a framework for anything serious, just a stupid one off christmas event for PAS.

**Comment by RemieRichards** [comment]
If it's for the event, it can be test merged over christmas.
that or you make it up to standards to remain in 24/7

**Comment by KorPhaeron** [comment]
I've long been against the idea of Valhalla since I think it'd encourage suicide whenever people don't get an antag role. I'd be very wary of it being in long term even if it met standards.

**Comment by Bawhoppen** [comment]
@KorPhaeron I never understood the arguement of it 'encouraging suicide' 
People said that about posibrains with malf but that never was an issue

**Comment by Shadowlight213** [comment]
@KorPhaeron technically the away mission antag spawners encourage suicide to the same level as Valhalla would. Since spawning in as an antag in an isolated Z level.
Valhalla would be just spawn in an arena and fight other ghosts rather than spawn in as an antag and fight the crew.

**Comment by duncathan** [comment]
Do you not want this to be merged?

**Comment by KorPhaeron** [comment]
@duncathan 

Was just going to test merge for christmas but I guess it wouldn't hurt to have around if other admins ever want to run it

**Pull request closed without merging**